### PR TITLE
[Serializer] Try all possible denormalization route with union types when ALLOW_EXTRA_ATTRIBUTES=false

### PR DESCRIPTION
**Pull request by T-bond**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | #45860
| License       | MIT
| Doc PR        | -

I found a similar bug, (I think this is a bug, but it is possible, it is an intended behaviour) to my previous report.

### CODE REVIEW

**Review by T-bond — COMMENTED** [review]
Self review done.

Failing tests seems unrelated.

File: src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php [hunk 1]
```diff
@@ -405,6 +405,7 @@ private function validateAndDenormalize(string $currentClass, string $attribute,
 
         $expectedTypes = [];
         $isUnionType = \count($types) > 1;
+        $extraAttributes = [];
```
Comment (NorthBlue333) [inline src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php]: In my opinion, `extraAttributes` should not be merged between different types.
I think the best would to save the first `ExtraAttributesException` thrown and throw it again if not null below.

**Comment by NorthBlue333** [comment]
Hello there,
I am currently working on #46284, and was wondering about the state of this PR. I made a comment because IMO we should not merge the extraAttributes of all different types, but just return one set failing.
Ideally this should be merged before my PR because I will have to adjust the code.

**Comment by T-bond** [comment]
This PR is ready for review.
If reviewed, and accepted I will rebase to resolve the conflicts.

File: src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php (hunk 1 above)
Comment (T-bond) [reply src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php]: I am not against it. The purpose of this PR to allow deserialization, when the first union does not contain a field provided in the serialized data, but one of the other union type is a perfect match.

Comment (dunglas) [reply src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php]: We could even the list of extra attributes per type in the exception message.

**Review by dunglas — APPROVED** [review]

File: src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php (hunk 1 above)
Comment (T-bond) [reply src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php]: I modified the code according to the suggestion of @NorthBlue333.

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @T-bond.

**Pull request merged**